### PR TITLE
Strict enforcement of chainId

### PR DESCRIPTION
**Pull request by GregTheGreek**
When opening a pull request to submit a new EIP, please use the suggested template: https://github.com/ethereum/EIPs/blob/master/eip-template.md

We have a GitHub bot that automatically merges some PRs. It will merge yours immediately if certain criteria are met:

 - The PR edits only existing draft PRs.
 - The build passes.
 - Your GitHub username or email address is listed in the 'author' header of all affected PRs, inside <triangular brackets>.
 - If matching on email address, the email address is the one publicly listed on your GitHub profile.

### CODE REVIEW

File: EIPS/eip-3788.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
+---
+eip: 3788
+title: Strict enforcement of chainId
+description: Strictly enforce the chainId on transactions based on the chain.
```
Comment (axic) [inline EIPS/eip-3788.md:4]: I think the abstract explains the goal better. I had no idea what this EIP does based on this line.

File: EIPS/eip-3788.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
+---
+eip: 3788
+title: Strict enforcement of chainId
+description: Reject transactions that do not explicitly have the same chainId as the node's configuration.
+author: Gregory Markou (@GregTheGreek)
+discussions-to: tbd
```
Comment (MicahZoltu) [inline EIPS/eip-3788.md:6]: This is a required field.  An ethereum-magicians forum thread is *strongly* recommended (you'll have to put up a bit of a fight to get other locations accepted).

**Review by MicahZoltu — CHANGES_REQUESTED** [review]
discussions-to is required.

----

Personal note (not a blocker for EIP), which I can add to the discussions-to once it is available:

On a separate note, I disagree with this change.  It doesn't add value to the protocol and it breaks backward compatibility.  Any wallet/signing tool that wants this can easily add it and we don't benefit from adding it to the core protocol that we cannot achieve by just convincing wallets/signers to add it.

In general, anything that can be solved at a higher layer than the core protocol I believe should be solved at a higher layer than the core protocol, and the core protocol should be as permissive as possible when given a choice between permissive and restrictive.

File: EIPS/eip-3788.md (hunk 2 above)
Comment (GregTheGreek) [reply EIPS/eip-3788.md]: Just got to my computer, added it. Thanks for the reminder.

File: EIPS/eip-3788.md [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+## Abstract
+
+Reject transactions that do not explicitly have the same chainId as the node's configuration.
+
+## Motivation
+
+Per [EIP-155](./eip-155.md) a transaction with a `chainId = 0` is considered to be a valid 
+transaction. This was a feature to offer developers the ability to sumbit replayable transactions 
+across different chains. With the rise of evm combatible chains, many of which use forks, or packages
```
Comment (MicahZoltu) [inline EIPS/eip-3788.md:22]:
```suggestion
across different chains. With the rise of evm compatible chains, many of which use forks, or packages
```

**Review by MicahZoltu — CHANGES_REQUESTED** [review]
Bot is blocking due to spelling error.

**Review by MicahZoltu — APPROVED** [review]

**Comment by sergiokessler** [comment]
> In general, anything that can be solved at a higher layer than the core protocol I believe should be solved at a higher layer than the core protocol, and the core protocol should be as permissive as possible when given a choice between permissive and restrictive.

can this be viewed as complete / incomplete info (instead of permissive / restrictive) ?

I mean, if there is the chainId the info is complete, incomplete otherwise.

**Comment by GregTheGreek** [comment]
> > In general, anything that can be solved at a higher layer than the core protocol I believe should be solved at a higher layer than the core protocol, and the core protocol should be as permissive as possible when given a choice between permissive and restrictive.
> 
> 
> 
> can this be viewed as complete / incomplete info (instead of permissive / restrictive) ?
> 
> 
> 
> I mean, if there is the chainId the info is complete, incomplete otherwise.
> 
> 

Please use the discussion link :)

**Pull request merged**